ignite/cmd/plugin.go

```diff
@@ -71,6 +76,108 @@ func linkPluginCmds(rootCmd *cobra.Command, p *plugin.Plugin) {
 	}
 }
 
+func linkPluginHooks(rootCmd *cobra.Command, p *plugin.Plugin) {
+	if p.Error != nil {
+		return
+	}
+
+	for _, hook := range p.Interface.Hooks() {
+		linkPluginHook(rootCmd, p, hook)
+		if p.Error != nil {
+			return
+		}
+	}
+}
+
+func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook plugin.Hook) {
+	cmdPath := hook.PlaceHookOn
+
+	if !strings.HasPrefix(cmdPath, "ignite") {
+		// cmdPath must start with `ignite ` before comparison with
+		// cmd.CommandPath()
+		cmdPath = "ignite " + cmdPath
+	}
+
+	cmdPath = strings.TrimSpace(cmdPath)
+
+	cmd := findCommandByPath(rootCmd, cmdPath)
+
+	if cmd == nil {
+		p.Error = errors.Errorf("unable to find commandPath %q for plugin hook %q", cmdPath, hook.Name)
+		return
+	}
+
+	if !cmd.Runnable() {
+		p.Error = errors.Errorf("can't attach plugin hook %q to non executable command %s", hook.Name, hook.PlaceHookOn)
+		return
+	}
+
+	preRun := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		errors := make([]string, 0)
+		var err error
+		if preRun != nil {
+			err = preRun(cmd, args)
+			if err != nil {
+				// dont return the error, log it and let execution continue to `Run`
+				errors = append(errors, err.Error())
```

Comment:
Good point for executing `preRun` before the plugin, I think it's better like that. But why don't you quit immediately in case of error ?

Reply:
Yea I think you're right here it should just fail. I was thinking that we should let the hook run but there isn't much point if the command aborts.